Library/Homebrew/cmd/tap.rb

```diff
@@ -58,12 +58,16 @@ def tap
     end
   end
 
+  def full_clone?
+    ARGV.include?("--full") || ARGV.homebrew_developer?
+  end
+
   # @deprecated this method will be removed in the future, if no external commands use it.
   def install_tap(user, repo, clone_target = nil)
```

Comment:
I think we could remove this function at this point. And inline `full_clone?` maybe?